src/Documentation/DocumentationGenerator/Extensions.cs

```diff
@@ -179,11 +179,37 @@ TypedExpression input
             // are an empty set; in that case, we want an empty string.
             ?? string.Empty;
 
-        internal static string ToSyntax(this QsCallable callable, Action<IRewriteStep.Diagnostic>? onDiagnostic = null) =>
-            SyntaxTreeToQsharp.DeclarationSignature(
+        internal static string ToSyntax(this QsFunctor functor) =>
+            functor.Tag switch
+            {
+                QsFunctor.Tags.Adjoint => "Adj",
+                QsFunctor.Tags.Controlled => "Ctl",
+                _ => "__invalid__"
+            };
+
+        internal static string ToSyntax(this QsCallable callable, Action<IRewriteStep.Diagnostic>? onDiagnostic = null)
+        {
+            var signature = SyntaxTreeToQsharp.DeclarationSignature(
                 callable, SyntaxTreeToQsharp.Default.ToCode
             );
 
+            // The signature provided by SyntaxTreeToQsharp doesn't include
+            // the characteristics (e.g.: `is Adj + Ctl`) for the callable, so
+            // we add that here.
+            var characteristics = callable.Signature.Information.Characteristics.SupportedFunctors switch
+            {
+                var nullable when nullable.IsNull => "",
+                { Item: var item } => item switch
+                {
+                    { Count: 0 } => "",
+                    // Be sure to add the leading space before is!
+                    _ => $" is {string.Join("+", item.Select(functor => functor.ToSyntax()))}"
+                }
```

Comment:
```suggestion
                { Item: { Count: 0 } } => "",
                // Be sure to add the leading space before is!
                { Item: var functors} => $" is {string.Join("+", functors.Select(functor => functor.ToSyntax()))}"
```

Reply:
For some reason I thought that nested pattern was a C# 9 feature... thanks for the suggestion!

Reply:
Looks like this change was not applied - did nested patterns turn out to be a C# 9 feature? (It worked in my IDE but I didn't try building it.)

Reply:
Ah, sorry, the web UI wouldn't allow me to accept the suggestion so I did it locally then forgot to push. Fixed now.